app_feup/lib/view/Widgets/NavigationDrawer.dart

```diff
@@ -25,18 +33,24 @@ class NavigationDrawerState extends State<NavigationDrawer> {
     "Bug Report"
   ];
 
+  getCurrentRoute() => ModalRoute.of(parentContext).settings.name == null ? 
+  drawerItems[0]:
+  ModalRoute.of(parentContext).settings.name.substring(1);
+
   _onSelectItem(int index) {
 
-    var prev = StoreProvider.of<AppState>(context).state.content["selected_page"];
+    final prev = getCurrentRoute();
 
-    Navigator.of(context).pop();
 
-    if (prev != drawerItems[index])  // If not already in selected page
-      Navigator.pushNamed(context, '/' + drawerItems[index]);
+    if (prev != drawerItems[index]){
+      Navigator.pushReplacementNamed(context, '/' + drawerItems[index]);
```

Comment:
Shouldn't it be Navigator.pushNamed? This way, when pressing the back, the screen goes black

Reply:
This was supposed to pop the navigator, it seems to not work. Will roll it back

Reply:
Nevermind, it was me who suggested it. Thing is pushReplacementNamed does not pop the drawer. It instead pops the whole page

Reply:
Yep, reverting back to poping and then pushing seems to solve the problem. I'm going to do the fix and then it's approved!

Reply:
Alright! Cool :shipit: